app/javascript/packages/verify-flow/hooks/use-secret-value.ts

```diff
@@ -0,0 +1,19 @@
+import { useContext, useState } from 'react';
+import SecretsContext from '../context/secrets-context';
+import type { SecretValues } from '../context/secrets-context';
+
+function useSecretValue<K extends keyof SecretValues>(
+  key: K,
+): [SecretValues[K], (nextValue: SecretValues[K]) => void] {
+  const store = useContext(SecretsContext);
+  const [value, setValue] = useState(store.getItem(key));
+
+  function setStateValue(nextValue: SecretValues[K]) {
+    store.setItem(key, nextValue);
+    setValue(nextValue);
```

Comment:
It occurs to me now as I submit this that this would only rerender the current instance which had set the value, not every consumer of the value. We probably want to track subscribers in a similar way we do with [`useCookie`](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/document-capture/hooks/use-cookie.js).

Reply:
>It occurs to me now as I submit this that this would only rerender the current instance which had set the value, not every consumer of the value. We probably want to track subscribers and notify them all, in a manner similar to what we do with [useCookie](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/document-capture/hooks/use-cookie.js).

Addressed as part of 80aa9f271. Rather than creating subscribers, let `setItem` trigger a change to the context value, which will cascade to all other consumers.